auth/services/selfsigned/types.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Nuts node
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package selfsigned
+
+import (
+	"encoding/json"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/auth/contract"
+	"github.com/nuts-foundation/nuts-node/vcr"
+)
+
+// ContractFormat is the contract format type
+const ContractFormat = "selfsigned"
+
+// VerifiablePresentationType is the dummy verifiable presentation type
+const VerifiablePresentationType = "NutsSelfSignedPresentation"
+
+// SessionCreated represents the session state after creation
+const SessionCreated = "created"
+
+// SessionInProgress represents the session state after rendering the html
+const SessionInProgress = "in-progress"
+
+// SessionCompleted represents the session state after the user has accepted the contract
+const SessionCompleted = "completed"
+
+type SessionStore interface {
```

Comment:
I find "Session" and "Store" low-level concepts (implementation details of services), especially given it doesn't have any "session" or "store"-esque functions (e.g. "startSession", "readX", "readY"), so I think this type should be named something like "SelfSignedMeans". But it's already in the `selfsigned` package, so just "Means" or "Service"?

Reply:
it's service now.